ie-tests/validate.py

```diff
@@ -0,0 +1,118 @@
+import os
+import re
+import tempfile
+import zipfile
+from argparse import ArgumentParser, Namespace
+
+from prettytable import PrettyTable
+
+
+def remove_ascii_colors(line):
+    """Clean the line from ascii color code."""
+    ansi_escape = re.compile(
+        r"""
+    \x1B  # ESC
+    (?:   # 7-bit C1 Fe (except CSI)
+        [@-Z\\-_]
+    |     # or [ for CSI, followed by a control sequence
+        \[
+        [0-?]*  # Parameter bytes
+        [ -/]*  # Intermediate bytes
+        [@-~]   # Final bytere
+    )
+""",
+        re.VERBOSE,
+    )
+    cleaned_line = ansi_escape.sub("", line.strip())
+    return cleaned_line
+
+
+def parse_args() -> Namespace:
+    """Parse command line args."""
+    parser = ArgumentParser()
+    parser.add_argument(
+        "-a", "--archive-logs", help="Path to the archive where run logs are stored."
+    )
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    # Read collections of logs
+    args = parse_args()
+
+    zip_file_folder = args.archive_logs
+    print("Start analyze the logs from the different runs...")
+    logs_archives = os.listdir(zip_file_folder)
+    log_results = {}
+    log_errors = {}
+    log_tests = {}
+    print(f"Number of runs: {len(logs_archives)}")
+    for log_archive in logs_archives:
```

Comment:
**suggestion** This for looks very ugly, and very nested. 

I would try to use generators to get something like:

```
@dataclass
class Result:
   run: int
   failures: int
   errors: int
   skipped: int
   raw: str

class Parser:
    # Parsing logic, we could have one for hadoop and one for spark
    @abstractmethod
    def parse(log_file:str) -> Result:
         ....

for log_file in extract_logs(zip_file_folder):
    result: Result = parser.parse(log_file)
    
    with open(f"/tmp/cleaned_{log_file}", "w") as f:
        f.write(result.raw)
```